### ISSUE TITLE
[Feature] Optionally Expose SQL Query to Show Unexpected Rows

### ISSUE THREAD

**Issue by jdimatteo** [fivetran/great_expectations#4181]
**Is your feature request related to a problem? Please describe.**

Sometimes it would be useful to have Great Expectations expose the SQL query so that a user can more quickly follow up on unexpected rows of data.

For some kinds of validation, a typical workflow is:
1) run Great Expectations to produce a Data Docs report
2) investigate expectation failures by querying the tables to better understand the context

Manually writing the query can sometimes be time consuming.

**Describe the solution you'd like**

It would save time if the query was included in the Data Docs report and the [ExpectationValidationResult](https://github.com/great-expectations/great_expectations/blob/56f63bfb11eeae0c0b8ed44b5fec7d039327f5eb/great_expectations/core/expectation_validation_result.py#L40), e.g.

![image](https://user-images.githubusercontent.com/3752732/153502997-313148e9-4d48-4d08-b52b-2894c7a1f121.png)

This might be configurable, e.g. using the [same example](https://github.com/jdimatteo/great_expectations/tree/feb_2022_feature_request_example/feature_example) detailed in #4170:

```
(venv) jdimatteo@erel:~/dev/great_expectations_fork_2/feature_example$ git diff -U7
diff --git a/feature_example/great_expectations_example.py b/feature_example/great_expectations_example.py
index f8caff28c..56b872ebc 100644
--- a/feature_example/great_expectations_example.py
+++ b/feature_example/great_expectations_example.py
@@ -55,14 +55,15 @@ suite = context.create_expectation_suite(expectation_suite_name=SUITE_AND_ASSET_
 suite.add_expectation(
     expectation_configuration=ExpectationConfiguration(
         expectation_type="expect_column_values_to_be_between",
         kwargs={
             "column": "age",
             "min_value": 0,
             "max_value": 150,
+            "include_unexpected_query": True,
         },
     )
 )
 context.save_expectation_suite(
     expectation_suite=suite, expectation_suite_name=SUITE_AND_ASSET_NAME
 )
```

**Describe alternatives you've considered**

We've considered using the [ExpectationValidationResult](https://github.com/great-expectations/great_expectations/blob/56f63bfb11eeae0c0b8ed44b5fec7d039327f5eb/great_expectations/core/expectation_validation_result.py#L40) `unexpected_list` and/or `unexpected_rows` to build up a query that includes the values but this is a round about and inefficient way to build/run a query and doesn't work for some kinds of expectations. 

**Additional context**

We intend to use this feature in combination with #4170 and #4185 and #4186, and the example here builds off the example detailed in #4170.

**Comment by austiezr** [comment]
Hey @jdimatteo ! Thanks for reaching out with these; a lot of super interesting functionality here. We'll review internally over the next week and continue the conversation.

**Comment by kyleaton** [comment]
Hey @jdimatteo, We have reviewed this item and added it to our feature roadmap. We do not have an estimated time to start work on this, but we will notify you when we do.

**Comment by talagluck** [comment]
Resolved by #6660
